lib/generators/arqo/named_base.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails/generators/named_base'
+
+module ARQO
+  module Generators
+    class NamedBase < Rails::Generators::NamedBase
+      argument :attributes, type: :array, default: [], banner: 'associated_to[:class_name]'
+
+      private
+
+      def associated_to
+        attributes.find { |attribute| attribute.name == 'associated_to' }&.type
+      end
+
+      def associated_relation
+        associated_to&.to_s&.classify
+      end
+
+      def factory
+        (associated_to || class_name)&.to_s&.underscore
```

Comment:
what about namespaces, does this work?

Reply:
If I run `rails g query Module::CustomUser associated_to:User`, this is the result:
```
RSpec.describe Module::CustomUserQuery do
  subject(:query) { described_class.new }

  describe '#all' do
    let!(:module_custom_users) { create_list(:user, 3) }

    it 'returns all Custom users' do
      expect(query.all).to match_array(module_custom_users)
    end
  end
end
```
Do you think it is right?

Reply:
looks good to me